pygis/docs/e_vector_overlay.md

```diff
@@ -0,0 +1,381 @@
+    print("\nFirst five rows:")
+    display(attribute_table.head())
+
+    # Display last five rows of attribute table
+    print("\nLast five rows:")
+    display(attribute_table.tail())
+
+
+def plot_overlay(overlay_type, overlay_result):
+    '''Plot the overlay result on a map and add the outlines of the original shapefiles on top.'''
+
+    # Create subplots
+    fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+
+    # Plot overlay data
+    overlay_result.plot(ax = ax, cmap = 'Set2', edgecolor = 'black')
+
+    # Plot outlines of original shapefiles
+    counties.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+    watersheds.plot(ax = ax, color = 'none', edgecolor = 'dodgerblue')
+
+    # Stylize plots
+    plt.style.use('bmh')
+
+    # Set title
+    ax.set_title('San Francisco Bay Area County and Watershed Boundaries\n{}'.format(overlay_type), fontdict = {'fontsize': '15', 'fontweight' : '3'})
+
+
+def plot_merge(merge_type, merge_result, merge_vector):
+    '''Plot the merge result on a map.'''
+
+    # Create subplots
+    fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+
+    # Plot data depending on vector type
+    # For points
+    if merge_vector == "point":
+
+        # Plot data
+        counties.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+        merge_result.plot(ax = ax, marker = 'o', color = 'dodgerblue', markersize = 3)
+
+    # For polygons
+    else:
+
+        # Plot data
+        merge_result.plot(ax = ax, cmap = 'Set2', edgecolor = 'black')
+
+    # Stylize plots
+    plt.style.use('bmh')
+
+    # Set title
+    ax.set_title('San Francisco Bay Area County Boundaries and Well Locations\n{}'.format(merge_type), fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+## Overlays
+
+For the first four, we can use the `overlay` function in `geopandas`. We simply change the argument for the `how` parameter to the overlay of our choosing.
+
+We will use the county boundaries and watershed boundaries shapefiles in these examples. The overlays will allow us to see what areas are only in a county, only in a watershed, or in both.
+
+Let's briefly examine the attribute table of our shapefiles and plot the data so that we know what we're working with.
+
+```{code-cell} ipython3
+# Create subplots
+fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+
+# Plot data
+counties.plot(ax = ax, color = 'bisque', edgecolor = 'dimgray', alpha = 0.75)
+watersheds.plot(ax = ax, color = 'lightskyblue', edgecolor = 'dodgerblue', alpha = 0.55)
+
+# Stylize plots
+plt.style.use('bmh')
+
+# Set title
+ax.set_title('San Francisco Bay Area County and Watershed Boundaries', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+```{code-cell} ipython3
+# Print attribute table
+display(counties)
+```
+
+```{code-cell} ipython3
+# Print attribute table
+display(watersheds)
+```
+
+### Union
+
+With `how='union'`, all data (all geometries regardless of overlap) is kept.
+
+```{figure} ../_static/img/vector_union.jpg
+:name: Union
```

Comment:
Good names are important. Use a short descriptive name for each image. Google SEO prefers sights with meaningful descriptions.

Reply:
Will update names. Let me know what you think!